src/query/sql/src/planner/binder/scalar_common.rs

```diff
@@ -227,17 +228,36 @@ pub fn wrap_cast_scalar(
     data_type: &DataType,
     target_type: &DataType,
 ) -> Result<ScalarExpr> {
-    let target_scalar = if data_type.remove_nullable() == DataType::String
-        && target_type.remove_nullable() == DataType::Variant
-    {
-        // parse string to JSON value
-        let func = ScalarExpr::FunctionCall(FunctionCall {
-            span: None,
-            func_name: "parse_json".to_string(),
-            params: vec![],
-            arguments: vec![scalar.clone()],
-        });
-        wrap_cast(&func, target_type)
+    let target_scalar = if target_type.remove_nullable() == DataType::Variant {
```

Comment:
Why need special case for Variant? `wrap_cast` should have supported varaiant

Reply:
`wrap_cast` will cast the string to a JSON string, not parse string to JSON value.

Reply:
If so, we should change the behavior of `to_variant(string)` instead of adding `if` here

Reply:
Changing `to_variant(string)` is indeed simpler. Currently, this behavior is meant to be compatible with Snowflake, I'm not sure if it should be changed, maybe we need to some discuss.

Reply:
`wrap_cast_scalar` should behave the same as `wrap_cast`. Since the `if` here, `CAST(... AS VARAIANT)` has been changed to `parse_json`, which means the compatibility has been broken with inconsistency.

Reply:
After discussing with @sundy-li , it's better to keep the current behavior for better compatibility. The `wrap_cast_scalar` function is only used for cast values in insert, copy and update. Casts to variant types use the `cast_scalar_to_variant` function, which doesn't convert to `parse_json`. So cast and insert values will behave differently. @andylokandy